common-custom-user-data-gradle-plugin/src/main/java/com/gradle/Utils.java

```diff
@@ -28,7 +29,11 @@ static boolean isNotEmpty(String value) {
         return value != null && !value.isEmpty();
     }
 
-    static Optional<String> sysProperty(String name) {
+    static Optional<String> sysProperty(String name, ProviderFactory providers) {
```

Comment:
This `Utils` file would likely make a nice library that we could give to plugin authors wanting to support both Gradle 5,6 and 7 builds and be compatible with the configuration cache.

Reply:
In the direction of a library, what might be particularly nice is to create a wrapper class (decorator) around ProviderFactory that handles the particulars of Gradle cross-version support.